storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -461,6 +466,75 @@ public SupervisorDetails getSupervisorById(String nodeId) {
         return networkTopography;
     }
 
+    /*
+    * Get heap memory usage for a worker's main process and logwriter process
+    * */
+    private Double getAssignedMemoryForSlot(Map topConf) {
+        Double totalWorkerMemory = 0.0;
+
+        String topologyWorkerChildopts = Utils.getString(topConf.get(Config.TOPOLOGY_WORKER_CHILDOPTS), null);
+        String workerChildopts = Utils.getString(topConf.get(Config.WORKER_CHILDOPTS), null);
```

Comment:
Both of these configs will return either a string or a list of strings.  We should support both options.